src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,36 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a type of Task that
+ * needs to be done by a set date and time.
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```

Comment:
Perhaps, a better variable name can be used here instead of by to make it more explicit that its the date/time for the deadline.